specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2023-03-01-preview/Capabilities.json

```diff
@@ -0,0 +1,562 @@
+          "readOnly": true
+        },
+        "serverCount": {
+          "type": "integer",
+          "description": "Count of servers in cache matching the spec",
+          "format": "int32",
+          "readOnly": true
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "../../../types/common-types.json#/definitions/CapabilityBase"
+        }
+      ]
+    },
+    "FlexibleServerEditionCapability": {
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "../../../types/common-types.json#/definitions/CapabilityBase"
+        }
+      ],
+      "properties": {
+        "name": {
+          "type": "string",
+          "description": "Server edition name",
+          "readOnly": true
+        },
+        "defaultSkuName": {
+          "type": "string",
+          "description": "Default sku name for the server edition",
+          "readOnly": true
+        },
+        "supportedStorageEditions": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/StorageEditionCapability"
+          },
+          "description": "The list of editions supported by this server edition.",
+          "readOnly": true,
+          "x-ms-identifiers": [
+            "name"
+          ]
+        },
+        "supportedServerSkus": {
+          "type": "array",
+          "description": "List of supported server SKUs.",
+          "items": {
+            "$ref": "#/definitions/ServerSku"
+          },
+          "x-ms-identifiers": [
+            "name"
+          ],
+          "readOnly": true
+        }
+      },
+      "description": "Flexible server edition capabilities."
+    },
+    "FlexibleServerCapability": {
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "../../../types/common-types.json#/definitions/CapabilityBase"
+        }
+      ],
+      "properties": {
+        "name": {
+          "type": "string",
+          "description": "Name of flexible servers capability"
+        },
+        "supportedServerEditions": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/FlexibleServerEditionCapability"
+          },
+          "x-ms-identifiers": [
+            "name"
+          ],
+          "readOnly": true,
+          "description": "List of supported flexible server editions"
+        },
+        "supportedServerVersions": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ServerVersionCapability"
+          },
+          "x-ms-identifiers": [
+            "name"
+          ],
+          "description": "The list of server versions supported for this capability.",
+          "readOnly": true
+        },
+        "fastProvisioningSupported": {
+          "type": "string",
+          "description": "Gets a value indicating whether fast provisioning is supported",
+          "readOnly": true,
+          "enum": [
+            "Enabled",
+            "Disabled"
```

Comment:
Why not call it Supported and NotSupported?

Reply:
This would require another api train to update existing property name "fastProvisioningSupported" to "fastProvisioning" as "Supported" or "NotSupported". Instead, we collectively decided to keep it as "fastProvisioningSupported" as "Enabled" / "Disabled". This api even though is exposed, is not for use by customers. This api is exposed mainly for it to be used from CLI to perform different operation checks what kind of provisioning is supported in different region.

Reply:
>This api is exposed mainly for it to be used from CLI to perform different operation checks what kind of provisioning is supported in different region.

That sounds like a flawed understanding of what 'used by customers' means to me. You're documenting it in the azure rest api specs, because it is there to be used by customers, and that is whether they want to use it via the CLI or any other way.

Reply:
Anyway - ultimately these suggestions about naming, I are making, to make the API intuitive, and I am just making them for the sake of trying to design the most consistent and easiest-to-understand for customers API for Azure that we can. You don't have to take these suggestions, but you should understand that this kind of thing ultimately impacts customer satisfaction.

Reply:
I agree with your suggestions, technically customers are not using this API, it is only used by Portal and CLI, only reason to include it in swagger to generate SDK to consume it in the CLI.

Reply:
> Anyway - ultimately these suggestions about naming, I are making, to make the API intuitive, and I am just making them for the sake of trying to design the most consistent and easiest-to-understand for customers API for Azure that we can. You don't have to take these suggestions, but you should understand that this kind of thing ultimately impacts customer satisfaction.

Sorry, we agree with your suggestions and they definitely help making the API more intuitively. However as Pradeep mentioned, this API is technically used only by Portal and CLI and hence we are moving forward by updating the description. To avoid these kind of issues, we will track and incorporate these review comments upfront during the API review internally while the API gets ready for implementation.